src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeAlleleCounts.java

```diff
@@ -652,26 +628,17 @@ GenotypeAlleleCounts copy() {
     @SuppressWarnings("unchecked")
     public <T extends Allele> List<T> asAlleleList(final List<T> allelesToUse) {
         Utils.nonNull(allelesToUse, "the input allele list cannot be null");
-        if (allelesToUse.size() < maximumAlleleIndex()) {
-            throw new IllegalArgumentException("the provided alleles to use does not contain an element for the maximum allele ");
-        }
+        Utils.validateArg(allelesToUse.size() >= maximumAlleleIndex(), "the provided alleles to use does not contain an element for the maximum allele");
         if (distinctAlleleCount == 1 ) {
-            if (ploidy == 1) {
-                return Collections.singletonList(allelesToUse.get(sortedAlleleCounts[0]));
-            } else {
-                return Collections.nCopies(ploidy, allelesToUse.get(sortedAlleleCounts[0]));
-            }
+            return Collections.nCopies(ploidy, allelesToUse.get(sortedAlleleCounts[0]));
         } else {
-            final Allele[] myAlleles = new Allele[ploidy];
-            int nextIndex = 0;
+            final List<T> result = new ArrayList<>(ploidy);
             for (int i = 0, ii = 0; i < distinctAlleleCount; i++) {
-                final Allele allele = allelesToUse.get(sortedAlleleCounts[ii++]);
+                final T allele = allelesToUse.get(sortedAlleleCounts[ii++]);
                 final int repeats = sortedAlleleCounts[ii++];
-                for (int j = 0; j < repeats; j++) {
-                    myAlleles[nextIndex++] = allele;
-                }
+                result.addAll(Collections.nCopies(repeats, allele));
```

Comment:
Using `Collections.nCopies()` is certainly cleaner, but adds an extra construction/garbage collection of a Collection object per iteration compared to the previous version. This is only worth caring about if this method is performance-sensitive, of course -- if it's not, then definitely prefer the cleaner version.